servo/connectors/kubernetes.py

```diff
@@ -1446,6 +1449,37 @@ async def get_pods(self) -> List[Pod]:
         pods = [Pod(p) for p in pod_list.items]
         return pods
 
+    async def get_latest_pods(self) -> List[Pod]:
+        """Get only the Deployment pods that belong to the latest ResourceVersion.
+
+        Returns:
+            A list of pods that belong to the latest deployment replicaset.
+        """
+        self.logger.info(f'getting replicaset for deployment "{self.name}"')
+        async with self.api_client() as api_client:
+            label_selector = self.obj.spec.selector.match_labels
+            rs_list:kubernetes_asyncio.client.V1ReplicasetList = await api_client.list_namespaced_replica_set(
+                namespace=self.namespace, label_selector=selector_string(label_selector), resource_version=self.resource_version
+            )
+
+        # Verify all returned RS have this deployment as an owner
+        rs_list = [
+            rs for rs in rs_list.items if rs.metadata.owner_references and any(
+                ownRef.kind == "Deployment" and ownRef.uid == self.obj.metadata.uid
+                for ownRef in rs.metadata.owner_references)]
+        if not rs_list:
+            raise servo.ConnectorError('Unable to locate replicaset(s) for deployment "{self.name}"')
+        latest_rs = sorted(rs_list, key= lambda rs: rs.metadata.resource_version, reverse=True)[0]
+
+        return [
+            pod for pod in await self.get_pods()
+            if any(
+                ownRef.kind == "ReplicaSet" and ownRef.uid == latest_rs.metadata.uid
+                for ownRef in pod.obj.metadata.owner_references
+            )]
```

Comment:
Also not the clearest read to my eye. Should this `any` check here and above be encapsulated into a method somewhere?

It's hard to tell what its actually trying to express even beyond the density of nested list comprehensions